src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    private static final int defaultSize = 10;
+    private Object[] arrayList = new Object[defaultSize];
+    private int size = 0;
+
+    private void grow() {
+        arrayList = Arrays.copyOf(arrayList, (int)(arrayList.length * 1.5));
+    }
+
     @Override
     public void add(T value) {
-
+        if (arrayList.length == size()) {
+            grow();
+        }
+        arrayList[size()] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size()) {
+            throw new ArrayListIndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+        if (size() == arrayList.length) {
+            grow();
+        }
+        Object[] newList = new Object[arrayList.length];
```

Comment:
Creating a new array `newList` and copying elements manually is inefficient. Consider using `System.arraycopy()` to copy array elements, as suggested in the checklist.